mlflow/utils/gorilla.py

```diff
@@ -788,7 +779,30 @@ def get_original_attribute(obj, name):
     --------
     :attr:`Settings.allow_hit`.
     """
-    return getattr(obj, _ORIGINAL_NAME % (name,))
+
+    original_name = _ORIGINAL_NAME % (name,)
+
+    def _get_attr(obj_, name_):
+        if bypass_descriptor_protocal:
+            return get_attribute(obj_, name_)
+        else:
+            return getattr(obj_, name_)
+
+    if inspect.isclass(obj):
+        for obj_ in inspect.getmro(obj):
+            if original_name in obj_.__dict__:
+                return _get_attr(obj_, original_name)
+            elif name in obj_.__dict__:
+                return _get_attr(obj_, name)
+            else:
+                # go on checking parent classes
+                continue
```

Comment:
Search from children classes to parent classes, and check "original_name" attribute first. This will ensure get the correct original attribute in any cases (e.g., the case some classes in the hierarchy haven't been patched, but some others are patched, this case the previous code is risky to get wrong original attribute)